### ISSUE TITLE
Make Simulation's PIMixin/CSVMixin set_timeseries and write similar to optimization

### ISSUE THREAD

**Issue by SGeeversAtVortech** [Deltares/rtc-tools#1127]
In GitLab by @vreeken on Oct 15, 2019, 15:48

Currently (in simulation)
- set_timeseries only works for PIMixin
- extract_results only works for PIMixin
- set_timeseries with output=True stores directly in timeseries_export, instead of relying on self.io (and getting it again later). The self.io call is superfluous (but shouldn't be).
- write of PIMixin's write loops only over values of `get_output_variables`, not those added with set_timeseries and output=True.
- Basically, it should be possible to use both CSVMixin and PIMixin together, that can read from each other's values (e.g. with extract_results)
